components/iam/pkg/oidc/service.go

```diff
@@ -97,43 +124,59 @@ func randString(size int) (string, error) {
 
 func (s *Service) GetClientConfigFromRequest(r *http.Request) (*ClientConfig, error) {
 	issuerParam := r.URL.Query().Get("issuer")
-	if issuerParam == "" {
-		return nil, errors.New("issuer param not specified")
+	stateParam := r.URL.Query().Get("state")
+	if issuerParam == "" && stateParam == "" {
+		return nil, fmt.Errorf("missing request parameters")
 	}
```

Comment:
I'd recommend doing 2 Ifs, and report which param is missing. It makes it a bit simpler when debugging

Reply:
That's an XOR gate and `GetClientConfigFromRequest` is used for /start and /callback with different params. 

It's a fair point though, the generalization would allow to pass this gate with unexpected params. It would make sense to split it up:
* `GetClientConfigFromStartRequest`
* `GetClientConfigFromCallbackRequest`